<launch>

    <!-- make shutter look forward? -->
    <arg name="look_forward" value="True"/>
    <arg name="publish_rate" default="30"/>
    <arg name="path_type" default="horizontal"/>
    <arg name="add_noise" default="false"/>
    <arg name="follow_target_topic" default="/filtered_target"/>

    <!-- bring up the robot -->
    <include file="$(find shutter_bringup)/launch/shutter.launch">
        <arg name="simulation" value="true"/>
    </include>

    <!-- generate the simulated moving object -->
    <node name="generate_continuous_target" pkg="shutter_kf" type="generate_continuous_target.py" output="screen">
	  <param name="publish_rate" value="$(arg publish_rate)"/>
      <param name="path_type" value="$(arg path_type)"/>
      <param name="add_noise" value="$(arg add_noise)"/>
    </node>

    <!-- visualize the moving object in rviz -->
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find shutter_lookat)/config/rviz_target.cfg.rviz"/>

    <!-- look forward if desired -->
    <group if="$(arg look_forward)">
        <node name="look_forward" pkg="shutter_lookat" type="look_forward.py"/>
    </group>

    <!-- run the filter -->
    <node name="kalman_filter" pkg="shutter_kf" type="kalman_filter.py" output="screen">
    </node>

    <!-- make the robot follow the filtered target -->
    <node name="expert_opt" pkg="shutter_behavior_cloning" type="expert_opt.py" output="screen">
        <param name="save_state_actions" value="False"/>
        <remap from="/target" to="$(arg follow_target_topic)"/>
    </node>

</launch>